airflow/providers/common/sql/operators/sql.py

```diff
@@ -1058,9 +1058,13 @@ def __init__(
 
     def execute(self, context: Context):
         hook = self.get_db_hook()
-        result = hook.get_first(self.sql)[0]
-        if not result:
+        result = hook.get_first(self.sql)
+
+        # if the query returns 0 rows result will be None so cannot be indexed into
+        if result is None:
```

Comment:
the result can be an empty list: https://github.com/apache/airflow/blob/dec2662190dd4480d0c631da733e19d2ec9a479d/airflow/providers/common/sql/hooks/sql.py#L396
```suggestion
        if not result:
```

Reply:
the `get_first` method on hook will only return the first row though right? It is calling `fetchone` on the result set from the DB.
So there will not be a list but a tuple if there is data or None if no data present from the query.

The issue with your change is that if the result of the query is `0` or `""` then the condition is trigger giving a false error, which is the context for this change.

Reply:
Ah yes, every operator has its own implementation. But your change is not correct, if an operator returns an empty list, dict, tuple, or string, `result[0]` will raise an exception.
What about using something like:
```python
if result is None or (isinstance([], Sized) and len(result) == 0)
```
here is the explanation:
```python
>>> from collections.abc import Sized
>>> isinstance([], Sized)
True
>>> isinstance((), Sized)
True
>>> isinstance({}, Sized)
True
>>> isinstance("", Sized)
True
>>> isinstance(0, Sized)
False
```

Reply:
Yes, that would do it in case the DB fetch changes to be able to return a collection in the future.
Empty str could be a false raise of this error but since the operator should only be dealing with numerics that should be ok. I will make the change along these lines then 👍

Reply:
After thinking some more,
the most surefire solution might be to try index in and catch the Index/Type error that will be raised if index 0 is out of bound for an empty list or a None type respectively.  The end result is similar and avoids multiple condition checks on the happy path that my other approach had.

```python
try:
    result = result[0]
except (TypeError, IndexError):
    ...
```
Let me know what you think.

Reply:
@hussein-awala does this look ok to? Any feedback?